src/server.ts

```diff
@@ -135,49 +141,85 @@ export class Server {
    *
-   * Allows you to directly inspect the current state of a contract. This is a backup way to access your contract data which may not be available via events or simulateTransaction.
+   * Allows you to directly inspect the current state of a contract. This is a
+   * backup way to access your contract data which may not be available via
+   * events or simulateTransaction.
    *
-   * @param {string} contractId - The contract ID containing the data to load. Encoded as Stellar Contract Address e.g. `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex string for backwards compatibility, but will likely be deprecated in the future.
+   * @param {string} contractId - The contract ID containing the data to load.
+   *    Encoded as Stellar Contract Address e.g.
+   *    `CCJZ5DGASBWQXR5MPFCJXMBI333XE5U3FSJTNQU7RIKE3P5GN2K2WYD5` or a hex
+   *    string for backwards compatibility, but will likely be deprecated in the
+   *    future.
    * @param {xdr.ScVal} key - The key of the contract data to load.
-   * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the {@link SorobanRpc.LedgerEntryResult} object with the current value.
+   * @param {string} [expirationType] - The "durability keyspace" that this
+   *    ledger key belongs to, which is either 'temporary' (the default) or
+   *    'persistent'.
+   *
+   * @returns {Promise<SorobanRpc.LedgerEntryResult>} Returns a promise to the
+   *    {@link SorobanRpc.LedgerEntryResult} object with the current value.
+   *
+   * @warning If the data entry in question is a 'temporary' entry, it's
+   *    entirely possible that it has expired out of existence. Future versions
+   *    of this client may provide abstractions to handle that.
    */
   public async getContractData(
     contractId: string,
     key: xdr.ScVal,
+    expirationType: 'temporary' | 'persistent' = 'temporary'
   ): Promise<SorobanRpc.LedgerEntryResult> {
-    const contractKey = xdr.LedgerKey.contractData(
+    let durability: xdr.ContractDataDurability;
+    switch (expirationType) {
+      case 'temporary':
+        durability = xdr.ContractDataDurability.temporary();
+        break;
+
+      case 'persistent':
+        durability = xdr.ContractDataDurability.persistent();
+        break;
+
+      default:
+        throw new TypeError(`invalid expirationType: ${expirationType}`);
+    }
+
+    let contractKey: string = xdr.LedgerKey.contractData(
       new xdr.LedgerKeyContractData({
-        contractId: new Contract(contractId).address().toBuffer(),
+        contract: new Contract(contractId).address().toScAddress(),
         key,
-      }),
+        durability,
+        bodyType: xdr.ContractEntryBodyType.dataEntry()   // expirationExtension is internal
+      })
     ).toXDR("base64");
-    const getLedgerEntriesResponse: SorobanRpc.GetLedgerEntriesResponse = await jsonrpc.post(
+
+    const getLedgerEntriesResponse = await jsonrpc.post<
+      SorobanRpc.GetLedgerEntriesResponse
+    >(
```

Comment:
Suggestion: Seems like we could simplify this with...
```ts
return this.getLedgerEntries([xdr.LedgerKey.contractData(...)])[0];
```

Reply:
I think the original intent was a better error on missing keys? (I didn't write this helper in the first place.) I cleaned it up based on your suggestion though, e.g. no reason to `await` when returning a `Promise`, anyway.